.github/workflows/check-tests.yaml

```diff
@@ -0,0 +1,41 @@
+name: Check integration tests workflow
+
+on: [pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-$${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  check-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            *.nix
+            go.*
+            **/*.go
+            integration_test/
+            config-example.yaml
+
+      - uses: DeterminateSystems/nix-installer-action@main
+        if: steps.changed-files.outputs.any_changed == 'true'
+      - uses: DeterminateSystems/magic-nix-cache-action@main
+        if: steps.changed-files.outputs.any_changed == 'true'
+
+      - name: Generate and check integration tests
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: |
+          nix develop --run "cd cmd/gh-action-integration-generator && go generate"
```

Comment:
I wonder if this needs to be:

```
nix develop --run -- "cd cmd/gh-action-integration-generator && go generate"
```

Reply:
noo its supposed to be `nix develop --command` I'm used to nix-shell where its `--run` sorry will fix right away

Reply:
fixed